ledger/src/blockstore_processor.rs

```diff
@@ -1106,21 +1110,28 @@ pub fn send_transaction_status_batch(
     statuses: Vec<TransactionExecutionResult>,
     balances: TransactionBalancesSet,
     token_balances: TransactionTokenBalancesSet,
-    inner_instructions: Vec<Option<InnerInstructionsList>>,
-    transaction_logs: Vec<TransactionLogMessages>,
+    mut inner_instructions: Vec<Option<InnerInstructionsList>>,
+    mut transaction_logs: Vec<TransactionLogMessages>,
     transaction_status_sender: TransactionStatusSender,
 ) {
     let slot = bank.slot();
-    if let Err(e) = transaction_status_sender.send(TransactionStatusBatch {
-        bank,
-        transactions: transactions.to_vec(),
-        iteration_order,
-        statuses,
-        balances,
-        token_balances,
-        inner_instructions,
-        transaction_logs,
-    }) {
+    if !transaction_status_sender.enable_cpi_and_log_storage {
+        inner_instructions = inner_instructions.iter().map(|_| None).collect();
+        transaction_logs = transaction_logs.iter().map(|_| vec![]).collect();
```

Comment:
I was curious to see if there was way to avoid these `collect()`'s. This was the best I could come up with: https://github.com/carllin/solana/commit/4bde604f1f4e633e2d8c0ed30e0372b396ae1362

Reply:
Oh hey, that's not bad! I tried something similar, but didn't think to use an Option for the `send()`. Thx!